src/builder/QuarkBuilder.sol

```diff
@@ -308,23 +295,33 @@ contract QuarkBuilder {
                 }
             }
 
-            plannedBridgeAmount += bridgeActionContext.amount;
+            if (Strings.stringEqIgnoreCase(bridgeActionContext.assetSymbol, transferIntent.assetSymbol)) {
+                transferTokenBridgeAmount += bridgeActionContext.amount;
+            } else {
+                // TODO: Specifically check that symbol is the payment.token
+                paymentTokenBridgeAmount += bridgeActionContext.amount;
+            }
         }
 
         // Verify transfer actions are affordable
         // NOTE: Assume all transfer actions are on the TransferIntent.chainId as Bridging logics is currently assuming destination at TransferIntent.chainId
         // NOTE: To support multi-chain transfers, call below functions to check repeatedly with each chainId and plannedBridgeAmount
         // for each chain (Likely passed from TransferIntent with a list of chain Id)
         assertTransferActionsAffordableOnTargetChain(
-            transferActions, chainAccountsList, transferIntent.chainId, plannedBridgeAmount
+            transferActions,
+            chainAccountsList,
+            transferIntent.chainId,
+            transferTokenBridgeAmount,
+            paymentTokenBridgeAmount
         );
     }
 
     function assertTransferActionsAffordableOnTargetChain(
         Actions.Action[] memory transferActions,
         Accounts.ChainAccounts[] memory chainAccountsList,
         uint256 targetChainId,
-        uint256 plannedBridgeAmountToTargetChain
+        uint256 transferTokenBridgeAmount,
+        uint256 paymentTokenBridgeAmount
```

Comment:
We actually need the amount bridged for both the transfer token and the payment token to do this check.

Reply:
I delete this function in #29, so one of our PRs is going to require a somewhat delicate rebase:

https://github.com/compound-finance/quark-scripts/pull/29/files#diff-9111113812791cbe6884c8fad436a6674788f2e44d1616c9bb124f30a33bd250L365